test/Microsoft.AspNet.Mvc.FunctionalTests/Compiler/Resources/MvcTagHelpersWebSite.MvcTagHelper_Home.CreateWarehouse.html

```diff
@@ -6,7 +6,7 @@
 <form action="/MvcTagHelper_Home/CreateWarehouse" method="post">        <div>
             <label class="warehouse" for="City">City</label>
             <input size="50" type="text" data-val="true" data-val-minlength="The field City must be a string or array type with a minimum length of &#39;2&#39;." data-val-minlength-min="2" id="City" name="City" value="" />
-            <span class="field-validation-valid" data-valmsg-for="City" data-valmsg-replace="true"></span>
+            <span class="field-validation-valid" data-valmsg-for="City" data-valmsg-replace="true" />
```

Comment:
don't do that.  stick to mainline scenarios in the functional tests.  and you're already testing correct handling of both self-closing and explicit end tag cases.

one other reason not to do this is you aren't testing anything about Razor or MVC, neither of which have the concept of "invalid to use explicit end tag with an element HTML defines as empty".